app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require "dry/monads"
+require "dry/monads/do"
+
+module Eligibilities
+  module AptcCsr
+    # This class applies eligibility overrides based on lawful presence
+    class ApplyEligibilityOverrides
+      include Dry::Monads[:result, :do]
+
+      # params are mm_app_with_member_determinations
+      def call(params)
+        result = yield apply_eligibility_overrides(params)
+        Success(result)
+      end
+
+      def apply_eligibility_overrides(params)
+        @mm_application = params[:magi_medicaid_application]
+        mm_app_hash = @mm_application.to_h
+        mm_app_hash[:tax_households].each do |mm_thh|
+          mm_thh[:tax_household_members].each do |thhm|
+            if only_medicaid_ineligible_due_to_immigration?(thhm) && (pregnancy_override?(thhm) || nineteen_to_twenty_one_override?(thhm))
+              thhm[:product_eligibility_determination][:is_magi_medicaid] = true
+            elsif only_chip_ineligible_due_to_immigration?(thhm) && under_eighteen_chip_override?(thhm)
+              thhm[:product_eligibility_determination][:is_medicaid_chip_eligible] = true
+            end
+          end
+        end
+        ::AcaEntities::MagiMedicaid::Operations::InitializeApplication.new.call(mm_app_hash)
+      end
+
+      def pregnancy_override?(thhm)
+        is_pregnant = applicant_by_reference(thhm[:applicant_reference][:person_hbx_id])&.pregnancy_information&.is_pregnant
+        is_pregnant && MedicaidGatewayRegistry[:eligibility_override].settings(:mitc_override_not_lawfully_present_pregnant).item
+      end
+
+      def nineteen_to_twenty_one_override?(thhm)
+        age = age(thhm)
+        nineteen_to_twenty_one = age > 18 && age < 21
+        nineteen_to_twenty_one && MedicaidGatewayRegistry[:eligibility_override].settings(:mitc_override_not_lawfully_present_under_twenty_one).item
+      end
+
+      def under_eighteen_chip_override?(thhm)
+        age(thhm) < 19 && MedicaidGatewayRegistry[:eligibility_override].settings(:mitc_override_not_lawfully_present_chip_eligible).item
+      end
+
+      def age(thhm)
```

Comment:
Calculating age dynamically could be problematic in some edge cases.  We should use the age stored on the application at the time of submission in EA.  There is a field in the applicant hash called `age_of_applicant` that can be used for this.